src/impute_srp_model.cpp

```diff
@@ -148,9 +148,29 @@ DataFrame impute_srp_model(
           }
         }
         if (from(i) == "response") {
-
-          // todo
-
+          if (from[i - 1] != "stable" ||
+              subject_id[i - 1] != subject_id[i]) {
+            stop("Unexpected");
+          }
+          double t_response = rtruncweibull(
```

Comment:
if you are coming "from response" there needs to be a stable-> response in the line before. You then sample response jump first (i-1) and then the progression jump (i). I now realize that this could be dangerous if the previous line is not correctly aligned in the data frame. Should probably add a check for that (e.g. that the subject ids are the same and the previous line is indeed a "from stable")